students/ZhuojingXie/lesson04/mailroom2.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+
+data = [{'name':'Andy','donation':[960,256,123.5,40]},{'name':'Bryce','donation':[30,45,27]},
+        {'name':'Charile','donation': [25,50]},{'name':'David', 'donation':[10]},
+        {'name':'Elaine','donation':[75,26]}]
+
+def menu_selection(prompt, dispatch_dict):
+    while True:
+        response = input(prompt)
+        if dispatch_dict[response]() == 'exit menu':
+            break
+
+def sub1_menu():
+    menu_selection(sub1_prompt,sub1_dispatch)
+
+def quit():
+    return 'exit menu'
+
+def thankletter(Name,Money):
+    print(f'\nDear {tuple[0]}, \n\n'
+         'Thank you so much for very generous donation of $'+f'{tuple[1]} to our chaple\n'
+         '\n\nRespectfully'
+         '\nZhuojing Xie')
+
+def list_name():
+    for info in data:
+        print(info['name'])
+    donator =input('\n\nPlease input the name or the donator > ')
+    money = input('\n\nThank you for donation , please type '
+                'the money you want to donate > ')
+    while not money.isdigit():
+        money = input('\n\nPlease type number only > ')
+    for info in data:
+        if donator == info['name']:
+            info['donation'].append(float(money))
+        else:
+            data.append({'name':donator,'donation':[float(money)]})
+            break
+    menu_selection(main_prompt,main_dispatch)
```

Comment:
you have a while True loop, you don't need to call `menu_selection`in all the functions, the loop should naturally repeat